var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -47,6 +48,7 @@ class Netcdf(Package):
 
     # Required for NetCDF-4 support
     depends_on("zlib")
+    depends_on("hdf5@:1.8.999", when='+hdf5_18')
```

Comment:
@eschnett AFAIK `:1.8` would mean everything up to **and including** 1.8. I recently looked into `Version` class and i think `1.8.xyz` satisfies `:1.8`. I would suggest to give it a quick try.